### DISCUSSION TITLE
feature: Handle "Warning: The file has been changed since reading it" natively

### DISCUSSION THREAD

**Discussion by murtaza64** [folke/noice.nvim#852] [Q&A]
### Did you check the docs?

- [X] I have read all the noice.nvim docs

### Is your feature request related to a problem? Please describe.

Currently, this message goes to my regular messages (which I personally use the `mini` view for):
<img width="875" alt="image" src="https://github.com/folke/noice.nvim/assets/13615693/a9ac1eb2-ad90-4b05-be3d-b0fef4f8beff">

### Describe the solution you'd like

It would be best if this was a user input prompt (i.e. in the middle of the screen), like what happens when you try to open a file that has a swapfile.

### Describe alternatives you've considered

I suppose I could set up a filter, but I'm not sure if that would handle getting user input since the message type seems to be `msg_show`

**Comment by murtaza64** [comment]
```
    routes = {
      {
        filter = { event = "msg_show", find = "Do you really want to" },
        view = "popup",
      },
    },
```
I tried stuff like this but it doesn't seem to work very well, it looks like this message isn't implemented like other input prompts in neovim (sorry I'm exhausted but I can provide a better description if you need one)

**Comment by cosmicbuffalo** [comment]
I've also been trying to figure out an elegant solution for this. I've been using the notify view for messages and I can't seem to get the lines for the prompt to merge into one notification. The best I've got so far is a route like this:
```lua
{
  view = "notify",
  filter = {
    event = "msg_show",
    ["not"] = { kind = { "confirm", "confirm_sub", "search_count" } },
  },
  opts = { replace = false, merge = true, title = "Merge Attempt" },
},
```
it at least prevents the notifications that pop up from replacing each other, but it still doesn't merge them together. This is what it spits out for me:

<img width="414" alt="Screenshot 2024-04-15 at 8 27 28 PM" src="https://github.com/folke/noice.nvim/assets/25158675/5a9cb295-d5fa-4ede-b621-49bd14a8e52e">

ignore that `return StatusCol(...)item` in there, I'm debugging that separately. Ideally the behavior I'm looking for from noice is just for those three notify bubbles to be combined into one, or to be turned into a confirm popup like the issue author suggests. Appreciate any help I can get.

**Comment by murtaza64** [comment]
If anyone knows who would be a good person to reach out to in neovim core, I would appreciate some help getting my first [PR](https://github.com/neovim/neovim/pull/29310) merged!
